core/trino-main/src/main/java/io/trino/operator/output/SlicePositionsAppender.java

```diff
@@ -82,38 +83,20 @@ public void append(IntArrayList positions, Block block)
         // performance of this method depends on block being always the same, flat type
         checkArgument(block instanceof VariableWidthBlock);
         ensurePositionCapacity(positionCount + positions.size());
-        int[] positionArray = positions.elements();
-        int newByteCount = 0;
-        int[] lengths = new int[positions.size()];
 
```

Comment:
We can make use of protected APIs by writing a method in BlockUtil and exposing that. It's not ideal but it might be better than adding the proposed new SPI methods from current PR.